geode-core/src/distributedTest/java/org/apache/geode/management/internal/cli/commands/AlterQueryServiceCommandWithSecurityDUnitTest.java

```diff
@@ -0,0 +1,256 @@
+          .requireNonNull(ClusterStartupRule.getCache()).getService(QueryConfigurationService.class)
+          .getMethodAuthorizer();
+
+      Set<String> expectedParameters =
+          new HashSet<>(Arrays.asList(JAVA_BEAN_AUTHORIZER_PARAMS.split(",")));
+      assertThat(methodAuthorizer.getAllowedPackages()).isEqualTo(expectedParameters);
+    }));
+
+    authorizerName = regExAuthorizerClass.getName();
+    gfsh.executeAndAssertThat(COMMAND_NAME + " --" + METHOD_AUTHORIZER_NAME + "=" + authorizerName
+        + " --" + SINGLE_AUTHORIZER_PARAMETER + "=" + REGEX_AUTHORIZER_PARAMETER).statusIsSuccess()
+        .hasTableSection().hasRowSize(serversToStart);
+
+    verifyCurrentAuthorizerClass(regExAuthorizerClass);
+    servers.forEach(server -> server.invoke(() -> {
+      RegExMethodAuthorizer methodAuthorizer = (RegExMethodAuthorizer) Objects
+          .requireNonNull(ClusterStartupRule.getCache()).getService(QueryConfigurationService.class)
+          .getMethodAuthorizer();
+
+      assertThat(methodAuthorizer.getAllowedPatterns().size()).isEqualTo(1);
+      assertThat(methodAuthorizer.getAllowedPatterns()).containsOnly(REGEX_AUTHORIZER_PARAMETER);
+    }));
+
+    authorizerName = restrictedAuthorizerClass.getName();
+    gfsh.executeAndAssertThat(COMMAND_NAME + " --" + METHOD_AUTHORIZER_NAME + "=" + authorizerName)
+        .statusIsSuccess().hasTableSection().hasRowSize(serversToStart);
+
+    verifyCurrentAuthorizerClass(restrictedAuthorizerClass);
+  }
+
+  @Test
+  public void alterQueryServiceCommandWithUserSpecifiedMethodAuthorizerUpdatesMethodAuthorizer()
+      throws IOException {
+    verifyCurrentAuthorizerClass(DEFAULT_AUTHORIZER_CLASS);
+
+    Class<TestMethodAuthorizer> authorizerClass = TestMethodAuthorizer.class;
+    String authorizerName = authorizerClass.getName();
+
+    String classContent =
+        new String(Files.readAllBytes(Paths.get(getFilePath(TEST_AUTHORIZER_TXT))));
+    String jarFileName = "testJar.jar";
+    File jarFile = tempFolder.newFile(jarFileName);
+    new ClassBuilder().writeJarFromContent(authorizerName, classContent, jarFile);
+
+    gfsh.executeAndAssertThat("deploy --jars=" + jarFile.getAbsolutePath()).statusIsSuccess();
+
+    gfsh.executeAndAssertThat(COMMAND_NAME + " --" + METHOD_AUTHORIZER_NAME + "=" + authorizerName
+        + " --" + AUTHORIZER_PARAMETERS + "=" + USER_AUTHORIZER_PARAMETERS).statusIsSuccess()
+        .hasTableSection().hasRowSize(serversToStart);
+
+    verifyCurrentAuthorizerClass(authorizerClass);
+    servers.forEach(server -> server.invoke(() -> {
+      TestMethodAuthorizer methodAuthorizer = (TestMethodAuthorizer) Objects
+          .requireNonNull(ClusterStartupRule.getCache()).getService(QueryConfigurationService.class)
+          .getMethodAuthorizer();
+
+      Set<String> expectedParameters =
+          new HashSet<>(Arrays.asList(USER_AUTHORIZER_PARAMETERS.split(",")));
```

Comment:
This might be able to be broken up into a test to validate that the parameters are split correctly.

Reply:
The parameters are split automatically by gfsh when it parses the `authorizer-parameters` argument. We split the input string here to generate the expected contents of the set of strings that is stored in the method authorizer rather than manually creating the set so that they match.

Reply:
A new test has been added to specifically verify that parameters are correctly set